packages/SwingSet/src/kernel/vatAdmin/vatAdminWrapper.js

```diff
@@ -21,37 +19,39 @@ export default function setup(syscall, state, helpers) {
   }
 
   function build(E, D) {
-    const vatIdsToRoots = new Map();
+    const vatIdsToResolvers = new Map();
 
     function createVatAdminService(vatAdminNode) {
       return harden({
         createVat(code) {
-          const vatId = D(vatAdminNode).create(code);
-          // If vat creation fails, we get the error message here, (Guaranteed
-          // not to start with a 'v'), so we don't create the promise or node.
-          if (!vatId.startsWith('v')) {
-            return vatId;
+          const result = D(vatAdminNode).create(code);
+          if (result.error) {
+            throw Error(`Vat Creation Error: ${result.error}`);
+          } else {
+            const vatPromise = makePromiseForVat();
```

Comment:
cosmetic thought: maybe use `makePromise` from `src/makePromise.js` instead of a local copy. I *think* it will import fine, it only depends upon `harden` and shouldn't drag anything else into the admin-vat's source string.